### PR TITLE
Fix TestCABundleChange test flake

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/48622/unit-tests_istio/1742433747514429440

Seems that the retry timeout is not long enough.

### CODE REVIEW

**Review by howardjohn — CHANGES_REQUESTED** [review]
hmm, on my machine I get flakes with this, but not without. 

```
=== RUN   TestCABundleChange
2024-01-03T15:51:40.648754Z	info	cluster "fake" kube client started
2024-01-03T15:51:40.651456Z	info	controllers	starting	controller=crd watcher
2024-01-03T15:51:40.680596Z	info	controllers	starting	controller=validation
2024-01-03T15:51:40.687637Z	info	validationController	Endpoint successfully rejected invalid config. Switching to fail-close.
2024-01-03T15:51:40.689460Z	info	validationController	successfully updated	name=istio-validator-revision-istio-system fail closed=true resource version= resource version=
2024-01-03T15:51:40.691352Z	info	validationController	successfully updated	name=istio-validator-revision-istio-system fail closed=true resource version= resource version=
2024-01-03T15:51:40.691761Z	info	validationController	up-to-date, no change required	name=istio-validator-revision-istio-system fail closed=true resource version=
2024-01-03T15:51:40.692220Z	info	validationController	up-to-date, no change required	name=istio-validator-revision-istio-system fail closed=true resource version=
2024-01-03T15:51:40.940034Z	info	validationController	successfully updated	name=istio-validator-revision-istio-system fail closed=true resource version= resource version=
2024-01-03T15:51:40.940544Z	info	validationController	up-to-date, no change required	name=istio-validator-revision-istio-system fail closed=true resource version=
    controller_test.go:299: found diff:   (*v1.ValidatingWebhookConfiguration)(Inverse(protocmp.Transform, protocmp.Message{
          	"@type": s"k8s.io.api.admissionregistration.v1.ValidatingWebhookConfiguration",
          	"Webhooks": []protocmp.Message{
        - 		s`{name:"hook0", clientConfig:{service:{namespace:"istio-system", name:"istio-revision", path:"/hook0"}, caBundle:"-----BEGIN CERTIFICATE-----\nMIIC9DCCAdygAwIBAgIJAIFe3lWPaalKMA0GCSqGSIb3DQEBCwUAMA4xDDAKBgNV\nBAMMA19jYTAgFw0xNzEyMjIxODA0MjRaGA8yMjkxMTAwNzE4`...,
        + 		s`{name:"hook0", clientConfig:{service:{namespace:"istio-system", name:"istio-revision", path:"/hook0"}, caBundle:"-----BEGIN CERTIFICATE-----\nMIIDCzCCAfOgAwIBAgIQbfOzhcKTldFipQ1X2WXpHDANBgkqhkiG9w0BAQsFADAv\nMS0wKwYDVQQDEyRhNzU5YzcyZC1lNjcyLTQwMzYtYWMzYy1k`...,
        - 		s`{name:"hook1", clientConfig:{service:{namespace:"istio-system", name:"istio-revision", path:"/hook1"}, caBundle:"-----BEGIN CERTIFICATE-----\nMIIC9DCCAdygAwIBAgIJAIFe3lWPaalKMA0GCSqGSIb3DQEBCwUAMA4xDDAKBgNV\nBAMMA19jYTAgFw0xNzEyMjIxODA0MjRaGA8yMjkxMTAwNzE4`...,
        + 		s`{name:"hook1", clientConfig:{service:{namespace:"istio-system", name:"istio-revision", path:"/hook1"}, caBundle:"-----BEGIN CERTIFICATE-----\nMIIDCzCCAfOgAwIBAgIQbfOzhcKTldFipQ1X2WXpHDANBgkqhkiG9w0BAQsFADAv\nMS0wKwYDVQQDEyRhNzU5YzcyZC1lNjcyLTQwMzYtYWMzYy1k`...,
          	},
          	"metadata": protocmp.Message{"@type": s"k8s.io.apimachinery.pkg.apis.meta.v1.ObjectMeta", "labels": map[string]string{"istio.io/rev": "revision"}, "name": string("istio-validator-revision-istio-system")},
          }))
        
        Left:  &ValidatingWebhookConfiguration{ObjectMeta:{istio-validator-revision-istio-system      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[istio.io/rev:revision] map[] [] [] []},Webhooks:[]ValidatingWebhook{ValidatingWebhook{Name:hook0,ClientConfig:WebhookClientConfig{Service:&ServiceReference{Namespace:istio-system,Name:istio-revision,Path:*/hook0,Port:nil,},CABundle:*[45 45 45 45 45 66 69 71 73 78 32 67 69 82 84 73 70 73 67 65 84 69 45 45 45 45 45 10 77 73 73 67 57 68 67 67 65 100 121 103 65 119 73 66 65 103 73 74 65 73 70 101 51 108 87 80 97 97 108 75 77 65 48 71 67 83 113 71 83 73 98 51 68 81 69 66 67 119 85 65 77 65 52 120 68 68 65 75 66 103 78 86 10 66 65 77 77 65 49 57 106 89 84 65 103 70 119 48 120 78 122 69 121 77 106 73 120 79 68 65 48 77 106 82 97 71 65 56 121 77 106 107 120 77 84 65 119 78 122 69 52 77 68 81 121 78 70 111 119 68 106 69 77 77 65 111 71 10 65 49 85 69 65 119 119 68 88 50 78 104 77 73 73 66 73 106 65 78 66 103 107 113 104 107 105 71 57 119 48 66 65 81 69 70 65 65 79 67 65 81 56 65 77 73 73 66 67 103 75 67 65 81 69 65 117 66 100 120 106 43 72 105 10 56 104 48 84 107 73 100 49 102 54 52 84 112 114 76 121 100 119 103 122 122 76 119 88 65 115 51 119 112 109 88 122 43 66 102 110 87 49 111 77 81 80 78 121 78 55 118 111 106 87 54 86 122 113 74 71 71 89 76 115 99 49 79 66 10 77 103 119 79 98 85 47 86 101 70 78 99 54 89 85 67 109 117 54 109 102 70 74 119 111 80 102 88 77 80 110 104 109 71 117 83 119 102 47 107 106 88 111 109 108 101 106 65 89 106 120 67 108 85 51 85 70 86 87 81 104 116 53 52 10 120 78 76 106 84 105 50 77 49 90 79 110 119 78 98 69 67 79 104 88 67 51 84 119 51 71 56 109 67 116 102 97 110 77 65 79 48 85 88 77 53 121 79 98 98 80 97 56 121 97 117 85 112 74 75 107 112 111 120 87 65 55 69 100 10 113 105 117 85 68 57 113 82 120 108 117 70 80 113 113 119 47 122 56 54 86 56 105 107 109 118 110 121 106 81 69 57 57 54 48 106 43 56 83 116 108 65 98 82 115 56 50 65 114 116 110 114 104 82 103 107 68 79 48 83 109 116 102 55 10 52 81 90 115 98 47 104 65 49 75 78 77 109 55 51 98 79 71 83 54 43 83 86 85 43 101 72 56 70 103 86 79 122 99 84 81 89 70 82 112 82 84 51 77 104 105 54 100 75 90 101 57 116 119 73 79 56 109 112 90 75 52 119 107 10 117 121 103 82 120 66 77 51 50 65 103 57 81 81 73 68 65 81 65 66 111 49 77 119 85 84 65 100 66 103 78 86 72 81 52 69 70 103 81 85 99 56 116 118 111 78 78 66 72 121 73 107 111 86 86 56 88 67 88 121 54 51 89 97 10 66 69 81 119 72 119 89 68 86 82 48 106 66 66 103 119 70 111 65 85 99 56 116 118 111 78 78 66 72 121 73 107 111 86 86 56 88 67 88 121 54 51 89 97 66 69 81 119 68 119 89 68 86 82 48 84 65 81 72 47 66 65 85 119 10 65 119 69 66 47 122 65 78 66 103 107 113 104 107 105 71 57 119 48 66 65 81 115 70 65 65 79 67 65 81 69 65 86 109 97 85 107 107 89 69 83 102 99 102 103 110 117 80 101 90 52 115 84 78 115 50 110 107 50 89 43 88 112 100 10 108 120 107 77 74 104 67 104 98 56 89 81 116 108 67 101 52 117 105 76 118 86 101 55 101 114 49 115 88 99 66 76 78 67 109 47 43 50 75 57 65 84 55 49 103 110 120 66 83 101 83 53 109 69 79 122 87 109 67 80 69 114 104 121 10 82 109 89 116 83 120 101 82 121 88 65 97 85 87 86 89 76 115 47 122 77 108 66 81 48 73 122 52 100 112 89 43 70 86 86 98 77 106 73 117 114 101 108 86 119 72 70 48 78 66 107 51 86 116 85 53 85 51 108 72 121 75 100 90 10 106 52 67 50 114 77 106 118 84 120 109 107 121 73 99 82 49 117 66 69 101 86 118 117 71 85 56 82 55 48 110 90 49 121 102 111 51 118 68 119 109 78 71 77 99 76 119 87 43 52 81 75 43 87 99 102 119 102 106 76 88 104 76 115 10 53 53 53 48 97 114 102 69 89 100 84 122 89 70 77 120 89 54 48 72 74 84 47 76 118 98 71 114 106 120 89 48 80 81 85 87 87 68 98 80 105 82 102 115 100 82 106 79 70 100 117 65 98 77 48 47 69 86 82 100 97 47 79 111 10 70 103 55 50 87 110 72 101 111 106 68 85 104 113 69 122 52 85 121 70 90 98 110 82 74 52 120 54 108 101 81 104 110 114 73 99 86 106 87 88 52 70 70 70 107 116 105 79 57 114 113 113 102 119 61 61 10 45 45 45 45 45 69 78 68 32 67 69 82 84 73 70 73 67 65 84 69 45 45 45 45 45],URL:nil,},Rules:[]RuleWithOperations{RuleWithOperations{Operations:[CREATE UPDATE],Rule:Rule{APIGroups:[group0],APIVersions:[*],Resources:[*],Scope:nil,},},},FailurePolicy:*Fail,NamespaceSelector:nil,SideEffects:nil,TimeoutSeconds:nil,AdmissionReviewVersions:[],MatchPolicy:nil,ObjectSelector:nil,MatchConditions:[]MatchCondition{},},ValidatingWebhook{Name:hook1,ClientConfig:WebhookClientConfig{Service:&ServiceReference{Namespace:istio-system,Name:istio-revision,Path:*/hook1,Port:nil,},CABundle:*[45 45 45 45 45 66 69 71 73 78 32 67 69 82 84 73 70 73 67 65 84 69 45 45 45 45 45 10 77 73 73 67 57 68 67 67 65 100 121 103 65 119 73 66 65 103 73 74 65 73 70 101 51 108 87 80 97 97 108 75 77 65 48 71 67 83 113 71 83 73 98 51 68 81 69 66 67 119 85 65 77 65 52 120 68 68 65 75 66 103 78 86 10 66 65 77 77 65 49 57 106 89 84 65 103 70 119 48 120 78 122 69 121 77 106 73 120 79 68 65 48 77 106 82 97 71 65 56 121 77 106 107 120 77 84 65 119 78 122 69 52 77 68 81 121 78 70 111 119 68 106 69 77 77 65 111 71 10 65 49 85 69 65 119 119 68 88 50 78 104 77 73 73 66 73 106 65 78 66 103 107 113 104 107 105 71 57 119 48 66 65 81 69 70 65 65 79 67 65 81 56 65 77 73 73 66 67 103 75 67 65 81 69 65 117 66 100 120 106 43 72 105 10 56 104 48 84 107 73 100 49 102 54 52 84 112 114 76 121 100 119 103 122 122 76 119 88 65 115 51 119 112 109 88 122 43 66 102 110 87 49 111 77 81 80 78 121 78 55 118 111 106 87 54 86 122 113 74 71 71 89 76 115 99 49 79 66 10 77 103 119 79 98 85 47 86 101 70 78 99 54 89 85 67 109 117 54 109 102 70 74 119 111 80 102 88 77 80 110 104 109 71 117 83 119 102 47 107 106 88 111 109 108 101 106 65 89 106 120 67 108 85 51 85 70 86 87 81 104 116 53 52 10 120 78 76 106 84 105 50 77 49 90 79 110 119 78 98 69 67 79 104 88 67 51 84 119 51 71 56 109 67 116 102 97 110 77 65 79 48 85 88 77 53 121 79 98 98 80 97 56 121 97 117 85 112 74 75 107 112 111 120 87 65 55 69 100 10 113 105 117 85 68 57 113 82 120 108 117 70 80 113 113 119 47 122 56 54 86 56 105 107 109 118 110 121 106 81 69 57 57 54 48 106 43 56 83 116 108 65 98 82 115 56 50 65 114 116 110 114 104 82 103 107 68 79 48 83 109 116 102 55 10 52 81 90 115 98 47 104 65 49 75 78 77 109 55 51 98 79 71 83 54 43 83 86 85 43 101 72 56 70 103 86 79 122 99 84 81 89 70 82 112 82 84 51 77 104 105 54 100 75 90 101 57 116 119 73 79 56 109 112 90 75 52 119 107 10 117 121 103 82 120 66 77 51 50 65 103 57 81 81 73 68 65 81 65 66 111 49 77 119 85 84 65 100 66 103 78 86 72 81 52 69 70 103 81 85 99 56 116 118 111 78 78 66 72 121 73 107 111 86 86 56 88 67 88 121 54 51 89 97 10 66 69 81 119 72 119 89 68 86 82 48 106 66 66 103 119 70 111 65 85 99 56 116 118 111 78 78 66 72 121 73 107 111 86 86 56 88 67 88 121 54 51 89 97 66 69 81 119 68 119 89 68 86 82 48 84 65 81 72 47 66 65 85 119 10 65 119 69 66 47 122 65 78 66 103 107 113 104 107 105 71 57 119 48 66 65 81 115 70 65 65 79 67 65 81 69 65 86 109 97 85 107 107 89 69 83 102 99 102 103 110 117 80 101 90 52 115 84 78 115 50 110 107 50 89 43 88 112 100 10 108 120 107 77 74 104 67 104 98 56 89 81 116 108 67 101 52 117 105 76 118 86 101 55 101 114 49 115 88 99 66 76 78 67 109 47 43 50 75 57 65 84 55 49 103 110 120 66 83 101 83 53 109 69 79 122 87 109 67 80 69 114 104 121 10 82 109 89 116 83 120 101 82 121 88 65 97 85 87 86 89 76 115 47 122 77 108 66 81 48 73 122 52 100 112 89 43 70 86 86 98 77 106 73 117 114 101 108 86 119 72 70 48 78 66 107 51 86 116 85 53 85 51 108 72 121 75 100 90 10 106 52 67 50 114 77 106 118 84 120 109 107 121 73 99 82 49 117 66 69 101 86 118 117 71 85 56 82 55 48 110 90 49 121 102 111 51 118 68 119 109 78 71 77 99 76 119 87 43 52 81 75 43 87 99 102 119 102 106 76 88 104 76 115 10 53 53 53 48 97 114 102 69 89 100 84 122 89 70 77 120 89 54 48 72 74 84 47 76 118 98 71 114 106 120 89 48 80 81 85 87 87 68 98 80 105 82 102 115 100 82 106 79 70 100 117 65 98 77 48 47 69 86 82 100 97 47 79 111 10 70 103 55 50 87 110 72 101 111 106 68 85 104 113 69 122 52 85 121 70 90 98 110 82 74 52 120 54 108 101 81 104 110 114 73 99 86 106 87 88 52 70 70 70 107 116 105 79 57 114 113 113 102 119 61 61 10 45 45 45 45 45 69 78 68 32 67 69 82 84 73 70 73 67 65 84 69 45 45 45 45 45],URL:nil,},Rules:[]RuleWithOperations{RuleWithOperations{Operations:[CREATE UPDATE],Rule:Rule{APIGroups:[group1],APIVersions:[*],Resources:[*],Scope:nil,},},},FailurePolicy:*Fail,NamespaceSelector:nil,SideEffects:nil,TimeoutSeconds:nil,AdmissionReviewVersions:[],MatchPolicy:nil,ObjectSelector:nil,MatchConditions:[]MatchCondition{},},},}
        Right: &ValidatingWebhookConfiguration{ObjectMeta:{istio-validator-revision-istio-system      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[istio.io/rev:revision] map[] [] [] []},Webhooks:[]ValidatingWebhook{ValidatingWebhook{Name:hook0,ClientConfig:WebhookClientConfig{Service:&ServiceReference{Namespace:istio-system,Name:istio-revision,Path:*/hook0,Port:nil,},CABundle:*[45 45 45 45 45 66 69 71 73 78 32 67 69 82 84 73 70 73 67 65 84 69 45 45 45 45 45 10 77 73 73 68 67 122 67 67 65 102 79 103 65 119 73 66 65 103 73 81 98 102 79 122 104 99 75 84 108 100 70 105 112 81 49 88 50 87 88 112 72 68 65 78 66 103 107 113 104 107 105 71 57 119 48 66 65 81 115 70 65 68 65 118 10 77 83 48 119 75 119 89 68 86 81 81 68 69 121 82 104 78 122 85 53 89 122 99 121 90 67 49 108 78 106 99 121 76 84 81 119 77 122 89 116 89 87 77 122 89 121 49 107 89 122 65 120 77 68 66 109 77 84 86 107 78 87 85 119 10 72 104 99 78 77 84 107 119 78 84 69 50 77 106 73 120 77 84 73 50 87 104 99 78 77 106 81 119 78 84 69 48 77 106 77 120 77 84 73 50 87 106 65 118 77 83 48 119 75 119 89 68 86 81 81 68 69 121 82 104 78 122 85 53 10 89 122 99 121 90 67 49 108 78 106 99 121 76 84 81 119 77 122 89 116 89 87 77 122 89 121 49 107 89 122 65 120 77 68 66 109 77 84 86 107 78 87 85 119 103 103 69 105 77 65 48 71 67 83 113 71 83 73 98 51 68 81 69 66 10 65 81 85 65 65 52 73 66 68 119 65 119 103 103 69 75 65 111 73 66 65 81 67 54 115 83 65 78 56 48 67 105 48 68 89 70 112 78 68 117 109 71 89 111 101 106 77 81 97 105 52 50 103 54 110 83 75 89 83 43 101 107 118 115 10 69 55 117 84 43 101 101 112 79 55 52 119 106 56 111 54 110 70 77 78 68 117 53 56 43 88 103 73 115 118 80 98 87 110 110 43 51 87 116 85 106 74 102 121 105 81 88 120 109 109 84 103 56 111 109 52 117 89 49 67 55 82 49 72 10 103 77 115 114 76 50 54 112 85 97 88 90 47 108 84 69 56 90 86 53 67 110 81 74 57 88 105 108 97 103 89 52 105 90 75 101 112 116 117 90 107 120 114 87 103 107 70 66 68 55 116 114 54 53 50 69 65 51 104 109 106 43 51 104 10 52 115 84 67 81 43 112 66 74 75 71 56 66 74 90 68 78 82 114 67 111 105 65 66 89 66 77 99 70 76 74 115 97 75 117 71 90 107 74 54 75 116 120 104 81 69 79 57 81 120 74 86 97 68 111 83 118 108 67 82 71 97 56 82 10 102 99 86 121 89 81 121 88 79 90 43 48 86 72 90 74 81 103 97 76 116 113 71 112 105 81 109 108 70 116 116 112 67 119 68 105 76 102 77 107 107 51 85 65 100 55 57 111 118 107 104 78 49 77 67 113 43 79 53 78 55 89 86 116 10 101 86 81 87 97 84 85 113 85 86 50 116 75 85 70 118 86 113 50 49 90 100 108 52 100 82 97 113 43 67 70 53 85 56 117 79 113 76 89 47 52 75 103 57 65 103 77 66 65 65 71 106 73 122 65 104 77 65 52 71 65 49 85 100 10 68 119 69 66 47 119 81 69 65 119 73 67 66 68 65 80 66 103 78 86 72 82 77 66 65 102 56 69 66 84 65 68 65 81 72 47 77 65 48 71 67 83 113 71 83 73 98 51 68 81 69 66 67 119 85 65 65 52 73 66 65 81 67 103 10 111 70 55 49 69 121 50 98 49 81 89 50 50 67 54 66 88 99 65 78 70 49 43 119 80 122 120 74 111 118 70 101 75 89 65 110 85 113 119 104 51 114 70 55 112 73 89 67 83 47 97 100 90 88 79 75 108 103 68 66 115 98 99 83 10 77 120 65 71 110 67 82 105 49 115 43 65 55 104 77 89 106 51 115 81 65 98 66 88 116 116 99 51 49 53 53 55 108 82 111 74 114 120 53 56 73 101 78 53 68 121 115 104 84 53 51 116 55 113 52 86 119 67 122 117 67 88 70 84 10 51 122 82 72 86 82 72 81 110 79 54 76 72 103 90 120 49 70 117 75 102 119 116 107 104 102 83 88 68 121 89 85 50 102 81 89 119 50 72 99 98 57 107 114 89 85 47 97 108 86 105 86 90 100 69 48 114 69 78 88 67 67 108 113 10 120 79 55 65 81 107 53 77 74 99 71 103 54 99 102 69 53 119 87 65 75 85 49 65 84 106 112 75 52 67 78 43 82 84 110 56 118 56 79 68 76 111 73 50 83 87 51 112 102 115 110 88 120 109 57 51 79 43 112 112 57 72 78 52 10 43 79 43 49 80 81 116 78 85 87 104 67 102 104 43 103 54 66 78 50 109 89 111 50 79 69 90 56 113 71 83 120 68 108 77 90 101 106 52 89 79 100 86 107 87 56 80 72 109 70 90 84 75 48 119 57 105 74 75 113 77 53 111 49 10 86 54 103 53 103 90 108 113 83 111 82 104 73 67 75 48 57 116 112 99 10 45 45 45 45 45 69 78 68 32 67 69 82 84 73 70 73 67 65 84 69 45 45 45 45 45],URL:nil,},Rules:[]RuleWithOperations{RuleWithOperations{Operations:[CREATE UPDATE],Rule:Rule{APIGroups:[group0],APIVersions:[*],Resources:[*],Scope:nil,},},},FailurePolicy:*Fail,NamespaceSelector:nil,SideEffects:nil,TimeoutSeconds:nil,AdmissionReviewVersions:[],MatchPolicy:nil,ObjectSelector:nil,MatchConditions:[]MatchCondition{},},ValidatingWebhook{Name:hook1,ClientConfig:WebhookClientConfig{Service:&ServiceReference{Namespace:istio-system,Name:istio-revision,Path:*/hook1,Port:nil,},CABundle:*[45 45 45 45 45 66 69 71 73 78 32 67 69 82 84 73 70 73 67 65 84 69 45 45 45 45 45 10 77 73 73 68 67 122 67 67 65 102 79 103 65 119 73 66 65 103 73 81 98 102 79 122 104 99 75 84 108 100 70 105 112 81 49 88 50 87 88 112 72 68 65 78 66 103 107 113 104 107 105 71 57 119 48 66 65 81 115 70 65 68 65 118 10 77 83 48 119 75 119 89 68 86 81 81 68 69 121 82 104 78 122 85 53 89 122 99 121 90 67 49 108 78 106 99 121 76 84 81 119 77 122 89 116 89 87 77 122 89 121 49 107 89 122 65 120 77 68 66 109 77 84 86 107 78 87 85 119 10 72 104 99 78 77 84 107 119 78 84 69 50 77 106 73 120 77 84 73 50 87 104 99 78 77 106 81 119 78 84 69 48 77 106 77 120 77 84 73 50 87 106 65 118 77 83 48 119 75 119 89 68 86 81 81 68 69 121 82 104 78 122 85 53 10 89 122 99 121 90 67 49 108 78 106 99 121 76 84 81 119 77 122 89 116 89 87 77 122 89 121 49 107 89 122 65 120 77 68 66 109 77 84 86 107 78 87 85 119 103 103 69 105 77 65 48 71 67 83 113 71 83 73 98 51 68 81 69 66 10 65 81 85 65 65 52 73 66 68 119 65 119 103 103 69 75 65 111 73 66 65 81 67 54 115 83 65 78 56 48 67 105 48 68 89 70 112 78 68 117 109 71 89 111 101 106 77 81 97 105 52 50 103 54 110 83 75 89 83 43 101 107 118 115 10 69 55 117 84 43 101 101 112 79 55 52 119 106 56 111 54 110 70 77 78 68 117 53 56 43 88 103 73 115 118 80 98 87 110 110 43 51 87 116 85 106 74 102 121 105 81 88 120 109 109 84 103 56 111 109 52 117 89 49 67 55 82 49 72 10 103 77 115 114 76 50 54 112 85 97 88 90 47 108 84 69 56 90 86 53 67 110 81 74 57 88 105 108 97 103 89 52 105 90 75 101 112 116 117 90 107 120 114 87 103 107 70 66 68 55 116 114 54 53 50 69 65 51 104 109 106 43 51 104 10 52 115 84 67 81 43 112 66 74 75 71 56 66 74 90 68 78 82 114 67 111 105 65 66 89 66 77 99 70 76 74 115 97 75 117 71 90 107 74 54 75 116 120 104 81 69 79 57 81 120 74 86 97 68 111 83 118 108 67 82 71 97 56 82 10 102 99 86 121 89 81 121 88 79 90 43 48 86 72 90 74 81 103 97 76 116 113 71 112 105 81 109 108 70 116 116 112 67 119 68 105 76 102 77 107 107 51 85 65 100 55 57 111 118 107 104 78 49 77 67 113 43 79 53 78 55 89 86 116 10 101 86 81 87 97 84 85 113 85 86 50 116 75 85 70 118 86 113 50 49 90 100 108 52 100 82 97 113 43 67 70 53 85 56 117 79 113 76 89 47 52 75 103 57 65 103 77 66 65 65 71 106 73 122 65 104 77 65 52 71 65 49 85 100 10 68 119 69 66 47 119 81 69 65 119 73 67 66 68 65 80 66 103 78 86 72 82 77 66 65 102 56 69 66 84 65 68 65 81 72 47 77 65 48 71 67 83 113 71 83 73 98 51 68 81 69 66 67 119 85 65 65 52 73 66 65 81 67 103 10 111 70 55 49 69 121 50 98 49 81 89 50 50 67 54 66 88 99 65 78 70 49 43 119 80 122 120 74 111 118 70 101 75 89 65 110 85 113 119 104 51 114 70 55 112 73 89 67 83 47 97 100 90 88 79 75 108 103 68 66 115 98 99 83 10 77 120 65 71 110 67 82 105 49 115 43 65 55 104 77 89 106 51 115 81 65 98 66 88 116 116 99 51 49 53 53 55 108 82 111 74 114 120 53 56 73 101 78 53 68 121 115 104 84 53 51 116 55 113 52 86 119 67 122 117 67 88 70 84 10 51 122 82 72 86 82 72 81 110 79 54 76 72 103 90 120 49 70 117 75 102 119 116 107 104 102 83 88 68 121 89 85 50 102 81 89 119 50 72 99 98 57 107 114 89 85 47 97 108 86 105 86 90 100 69 48 114 69 78 88 67 67 108 113 10 120 79 55 65 81 107 53 77 74 99 71 103 54 99 102 69 53 119 87 65 75 85 49 65 84 106 112 75 52 67 78 43 82 84 110 56 118 56 79 68 76 111 73 50 83 87 51 112 102 115 110 88 120 109 57 51 79 43 112 112 57 72 78 52 10 43 79 43 49 80 81 116 78 85 87 104 67 102 104 43 103 54 66 78 50 109 89 111 50 79 69 90 56 113 71 83 120 68 108 77 90 101 106 52 89 79 100 86 107 87 56 80 72 109 70 90 84 75 48 119 57 105 74 75 113 77 53 111 49 10 86 54 103 53 103 90 108 113 83 111 82 104 73 67 75 48 57 116 112 99 10 45 45 45 45 45 69 78 68 32 67 69 82 84 73 70 73 67 65 84 69 45 45 45 45 45],URL:nil,},Rules:[]RuleWithOperations{RuleWithOperations{Operations:[CREATE UPDATE],Rule:Rule{APIGroups:[group1],APIVersions:[*],Resources:[*],Scope:nil,},},},FailurePolicy:*Fail,NamespaceSelector:nil,SideEffects:nil,TimeoutSeconds:nil,AdmissionReviewVersions:[],MatchPolicy:nil,ObjectSelector:nil,MatchConditions:[]MatchCondition{},},},}
--- FAIL: TestCABundleChange (2.32s)
2024-01-03T15:51:42.924075Z	info	controllers	stopped	controller=crd watcher
2024-01-03T15:51:42.924218Z	info	controllers	stopped	controller=validation
FAIL

ERROR: exit status 1
```

**Comment by howardjohn** [comment]
I think we do 2 updates because we reui-enque when we become ready ?

**Comment by hanxiaop** [comment]
> I think we do 2 updates because we reui-enque when we become ready ?

Yes, we do requeue, but when there is no change in the webhook, it will not be updated again. So should be 1 update for fail and 1 update for caBundle change.

After rechecking the test, I haven't seen any obvious problems. I think this PR can just be closed, or maybe increase the `LongRetry` slightly?

**Comment by hanxiaop** [comment]
BTW I guess the flake you've seen could be the inconsistent get behavior of the test client. What I observed is that, if the webhook is updated successfully without error, in a new reconcile loop, sometimes `webhooks.get` returns the updated one and sometimes it returns the un-updated one, as can also be seen in the logs:
```
=== RUN   TestCABundleChange
...
2024-01-04T05:34:11.706300Z	info	validationController	Endpoint successfully rejected invalid config. Switching to fail-close.
2024-01-04T05:34:11.706412Z	info	validationController	successfully updated	name=istio-validator-revision-istio-system fail closed=true resource version= resource version=
2024-01-04T05:34:11.706442Z	info	validationController	up-to-date, no change required	name=istio-validator-revision-istio-system fail closed=true resource version=
...
```
```
=== RUN   TestCABundleChange
...
2024-01-04T05:54:09.093553Z	info	validationController	Endpoint successfully rejected invalid config. Switching to fail-close.
2024-01-04T05:54:09.093577Z	info	validationController	successfully updated	name=istio-validator-revision-istio-system fail closed=true resource version= resource version=
2024-01-04T05:54:09.093601Z	info	validationController	successfully updated	name=istio-validator-revision-istio-system fail closed=true resource version= resource version=
2024-01-04T05:54:09.093673Z	info	validationController	up-to-date, no change required	name=istio-validator-revision-istio-system fail closed=true resource version=
...
--- PASS: TestCABundleChange (0.08s)
```

The second one updated the webhook twice because `webhooks.get` gets the old object even after update is successful.

File: pkg/webhooks/validation/controller/controller_test.go [hunk 1]
```diff
@@ -307,7 +307,7 @@ func TestCABundleChange(t *testing.T) {
 
 // LongRetry is used when comparing webhook values. Apparently the values are so large that with -race
 // on the comparison can take a few seconds, meaning we never retry with the default settings.
-var LongRetry = retry.Timeout(time.Second * 1)
+var LongRetry = retry.Timeout(time.Second * 2)
```
Comment (howardjohn) [inline pkg/webhooks/validation/controller/controller_test.go:310]:
```suggestion
var LongRetry = retry.Timeout(time.Second * 20)
```

It used to be 20, accidentally was changed to 1

**Review by howardjohn — APPROVED** [review]

**Pull request merged**